modules/core/src-scala-3/SourceLocationMacro.scala

```diff
@@ -25,7 +25,7 @@ object macros {
 
     val position = Position.ofMacroExpansion
 
-    val psj = position.sourceFile.jpath
+    val psj = position.sourceFile.getJPath.get
```

Comment:
Source: https://github.com/lampepfl/dotty/blob/7e0689996e0fd62012ba4982a4a6aefa23518c66/library/src/scala/quoted/Quotes.scala#L4240-L4247

I don't think `.get` makes this less safe, as previously it could return null (for virtual files) and would blow up anyways.

May be we need to turn SourceLocation into ADT, but I worry it's a breaking change for very little benefit.